cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -225,7 +225,12 @@ function move_sriov_pfs_netns_to_node {
   mkdir -p /var/run/netns/
   ln -sf /proc/$pid/ns/net "/var/run/netns/$node"
 
-  local -r sriov_pfs=( /sys/class/net/*/device/sriov_numvfs )
+  local -r sriov_pfs=( $(find /sys/class/net/*/device/sriov_numvfs) )
```

Comment:
Recently we removed all the error masking (removing `||return/exit`) and relay on `-e` flag
This change actually masking the error from `find` in order to echo more informative one,
the intention is good I like informative error message.
I dont like that we getting back to masking error even for better error message, where we can achive that w/o masking the error.
Instead of this check I prefer that we echo before this line "Discovering PF's" or something like that.
In case of an error the script will stop, and in the logs we could see where it failed 
and why because the error from `find` will reflect in the logs.
What you think?

Reply:
This is not this change, its the local that mask it, and we do need local imo.
This change is just about changing to find instead the bash globbing.
I dont think we should echo and search, code should do it for us.

Another option is remove the local here, but someone might add it back.
(we will need to keep the find in both cases, as the bash glob won't error in case of an error)

Reply:
@ormergi , IIUC you refer to the `find` execution as if it can fail and we should care.
I think that in this specific case, it is used to just fetch the list of PF/s and for this context it does not matter if it passes or fails.. the only thing that matters is the return value. Just think of it as if it fails, it returns an empty list.

If you look at it this way, we do not need to catch any errors here. And the error is only about the list content (if it is empty, we exit early).

Is this perspective better fits what is going on here?

Reply:
You can declare `sriov_pfs` and then do the assignment
```
local sriov_pfs
sirov_pfs=( $(find /sys/class/net/*/device/sriov_numvfs) )
```
That way the error is not masked

Reply:
> I dont think we should echo and search, code should do it for us.

I am not sure what you mean, at the current state I assume you added a check because the error was masked, with what I suggested you can unmask and remove that check

Reply:
Its nice, but its prone that someone will merge the lines, and then mask the error,
it might be fine even if it happens, because the code later will catch it, but i think we better fail fast, 
and reduce noises, wdyt?
also it should be `local -r`, once we get an agreement here i will try it out accordingly

Reply:
The point is that the error does not matter. We do not want to exit on `find` failing, we want to exit on the list being empty.

Same as if you would have this function: `get_pf_count()`. You will not expect it to fail, just to return a list/array.

Reply:
Well in general the code is prune to someone who is going to do something wrong, 
we hope that we or a maintainer will catch it during reviews.
I think it doesn't need to contradict with "doing things the right way", maybe I am too optimistic that also an option.
We could add a comment about dividing it to two lines, about using `-r` I think it is not a must here.

About the error, I cant think of a reason for `find` to fail other than file doesn't exists or missing permissions
in both cases the script should stop at that point because it means there are no PF's on the host default netns.
In this case there I dont think we benefit from saving `find` output to array and check if its empty, its just variables substitution. Its also requiring another error checking instead of relaying on `-e` flag.

In any case if this is going to stay consider changing `exit 1` to `return 1`

Reply:
> In any case if this is going to stay consider changing `exit 1` to `return 1`

IIRC when i debugged it, i saw that we can't, because `NODE_PFS=($(move_sriov_pfs_netns_to_node "$SRIOV_NODE" "$PF_COUNT_PER_NODE"))` 
the caller is also scoped.

Reply:
@EdDev Sorry I missed you comment, please see the one above
About find failing, if it fails it means that there are not PF's and the script stops, so I think it is redundant to save its output in array and check if its empty.

Reply:
> In any case if this is going to stay consider changing `exit 1` to `return 1`

Done

Reply:
You are looking at "find" failure, and I am looking at unifying results.
Lets take the "list_pfs" hypothetical example function. We could allow it to fail/raise in case something is wrong, or just return an empty list. The advantage of getting an empty list is that we can handle it the same way as a non-empty list.
This means we could iterate over it, check its length, etc...

I think this is the same case here... we are not interested in finding out that "find" failed, we are interested in checking its output. If something fails, you do not really want to know that "find" failed, you want to know that there are no PF/s without entering the details on how it was detected.

It is also worth mentioning that this is too minor to deserve such a long discussion. If it is not wrong, it is just about taste and there are strong opinions, someone needs to back-off in order to get forward. It is not that important to be blocked by either a reviewer or an author. You both should consider this for the next time.

Reply:
My point is that in this case there is no benefit in saving `find` output in an array and check its length 
because `find` could fail only if a given path doesn't exists (this script runs as sudo on CI so file permissions is not relevant)
If `find` fails the array is empty, if it doesn't fail the array is not empty.

About the long discussion, noted and appreciated.